contrib/epee/src/hex.cpp

```diff
@@ -83,4 +83,67 @@ namespace epee
   {
     return write_hex(out, src);
   }
+
+  std::vector<uint8_t> from_hex::vector(const std::string &src)
+  {
+    // should we ignore a specific character
+    auto ignore = [](char input) {
+        // we ignore spaces and colons
+        return std::isspace(input) || input == ':';
+    };
+
+    // the number of relevant characters to decode
+    auto count = std::count_if(src.begin(), src.end(), ignore);
+
+    // this must be a multiple of two, otherwise we have a truncated input
+    if (count % 2) {
+      throw std::length_error{ "Invalid hexadecimal input length" };
+    }
```

Comment:
That seems to be mean "if the number of ignored characters is odd, throw". Should it not be something like: if ((src.size()-count)%2) ?